tests/components/rest/test_sensor.py

```diff
@@ -1,696 +1,665 @@
+            }
+        },
+    )
+
+    await hass.async_block_till_done()
+    assert len(hass.states.async_all()) == 1
+
+
+@respx.mock
+async def test_setup_get_digest_auth(hass):
+    """Test setup with valid configuration."""
+    respx.get("http://localhost", status_code=200, content="{}")
+    assert await async_setup_component(
+        hass,
+        "sensor",
+        {
+            "sensor": {
+                "platform": "rest",
+                "resource": "http://localhost",
+                "method": "GET",
+                "value_template": "{{ value_json.key }}",
+                "name": "foo",
+                "unit_of_measurement": DATA_MEGABYTES,
+                "verify_ssl": "true",
+                "timeout": 30,
+                "authentication": "digest",
+                "username": "my username",
+                "password": "my password",
+                "headers": {"Accept": CONTENT_TYPE_JSON},
+            }
+        },
+    )
+
+    await hass.async_block_till_done()
+    assert len(hass.states.async_all()) == 1
+
+
+@respx.mock
+async def test_setup_post(hass):
+    """Test setup with valid configuration."""
+    respx.post("http://localhost", status_code=200, content="{}")
+    assert await async_setup_component(
+        hass,
+        "sensor",
+        {
+            "sensor": {
+                "platform": "rest",
+                "resource": "http://localhost",
+                "method": "POST",
+                "value_template": "{{ value_json.key }}",
+                "payload": '{ "device": "toaster"}',
+                "name": "foo",
+                "unit_of_measurement": DATA_MEGABYTES,
+                "verify_ssl": "true",
+                "timeout": 30,
+                "authentication": "basic",
+                "username": "my username",
+                "password": "my password",
+                "headers": {"Accept": CONTENT_TYPE_JSON},
+            }
+        },
+    )
+    await hass.async_block_till_done()
+    assert len(hass.states.async_all()) == 1
+
+
+@respx.mock
+async def test_setup_get_xml(hass):
+    """Test setup with valid xml configuration."""
+    respx.get(
+        "http://localhost",
+        status_code=200,
+        headers={"content-type": "text/xml"},
+        content="<dog>abc</dog>",
+    )
+    assert await async_setup_component(
+        hass,
+        "sensor",
+        {
+            "sensor": {
+                "platform": "rest",
+                "resource": "http://localhost",
+                "method": "GET",
+                "value_template": "{{ value_json.dog }}",
+                "name": "foo",
+                "unit_of_measurement": DATA_MEGABYTES,
+                "verify_ssl": "true",
+                "timeout": 30,
+            }
+        },
+    )
+    await hass.async_block_till_done()
+    assert len(hass.states.async_all()) == 1
+
+    state = hass.states.get("sensor.foo")
+    assert state.state == "abc"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)